src/components/App/ImportStudyList.js

```diff
@@ -0,0 +1,168 @@
+                            )
+                    )
+                ).forEach((response) => {
+                    response.schools
+                        .map((school) => school.departments)
+                        .flat()
+                        .map((dept) => dept.courses)
+                        .flat()
+                        .forEach((course) => {
+                            course.sections.forEach((section) => {
+                                addCourse(section, course, this.state.selectedTerm, currSchedule);
+                                ++sectionsAdded;
+                            });
+                        });
+                });
+                if (sectionsAdded === sectionCodes.length) {
+                    openSnackbar('success', `Successfully imported ${sectionsAdded} of ${sectionsAdded} classes!`);
+                } else if (sectionsAdded !== 0) {
+                    openSnackbar(
+                        'warning',
+                        `Successfully imported ${sectionsAdded} of ${sectionCodes.length} classes. 
+                        Please make sure that you selected the correct term and that none of your classes are missing.`
+                    );
+                } else {
+                    openSnackbar(
+                        'error',
+                        'Failed to import any classes! Please make sure that you pasted the correct Study List.'
+                    );
+                }
+            } catch (e) {
+                this.handleError(e);
+            }
+        }
+    };
+
+    enterEvent = (event) => {
+        const charCode = event.which ? event.which : event.keyCode;
+        if (charCode === 13 || charCode === 10) {
+            event.preventDefault();
+            this.handleClose(true);
+        }
+    };
+
+    componentDidUpdate(prevProps, prevState, snapshot) {
+        if (!prevState.isOpen && this.state.isOpen) {
+            document.addEventListener('keydown', this.enterEvent, false);
+        } else if (prevState.isOpen && !this.state.isOpen) {
+            document.removeEventListener('keydown', this.enterEvent, false);
+        }
+    }
+
+    render() {
+        return (
+            <>
+                <Button onClick={this.handleOpen} color="inherit" startIcon={<GetApp />}>
+                    Import
+                </Button>
+                <Dialog open={this.state.isOpen}>
+                    <DialogTitle>Import Schedule</DialogTitle>
+                    <DialogContent>
+                        <DialogContentText>
+                            Paste the contents of your Study List below to import it into AntAlmanac. Make sure you have
+                            the right term selected below.
+                        </DialogContentText>
+                        <InputLabel>Term</InputLabel>
+                        <Select value={this.state.selectedTerm} onChange={this.handleChange}>
+                            {termData.map((term) => (
```

Comment:
This causes a console warning because each MenuItem has no unique key prop. Get the index param from map and pass it to each menu item as a key prop (See `CustomAppBar.js` lines 81-83 for a good example of this).